test/uk/gov/hmrc/tai/repositories/IncomeRepositorySpec.scala

```diff
@@ -119,6 +118,20 @@ class IncomeRepositorySpec extends PlaySpec with MockitoSugar {
           Seq(bankAccount, bankAccount)))
       }
 
+      "non-tax-code income without untaxed interest is present" in {
```

Comment:
Could move this test into a dedicated block, to clarify your new functionality, e.g. something like ..
must "bypass any bank account retrieval and return no untaxed interest" 
when "no UntaxedInterestIncome is present"
.. and you could add an assertion that the bbsi repo wasn't invoked. I know we wouldn't normally have internal call checking like this, but since it's the specific nature of your change I think it seem acceptable here.